resources/Microsoft.VSCode.DSC/Microsoft.VSCode.DSC.psm1

```diff
@@ -0,0 +1,125 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+$ErrorActionPreference = "Stop"
+Set-StrictMode -Version Latest
+
+enum Ensure
+{
+    Absent
+    Present
+}
+
+#region DSCResources
+[DSCResource()]
+class VSCodeExtension
+{
+    [DscProperty(Key)]
+    [string] $Name
+
+    [DscProperty()]
+    [string] $Version
+
+    [DscProperty()]
+    [Ensure] $Ensure = [Ensure]::Present
+
+    [DscProperty(NotConfigurable)]
+    [hashtable] $InstalledExtensions = @{}
+
+    [VSCodeExtension] Get()
+    {
+        Assert-VSCode
+
+        $currentState = [VSCodeExtension]::new()
+        $currentState.Ensure = [Ensure]::Absent
+        $currentState.Name = $this.Name
+        $currentState.Version = $this.Version
+        $extensionList = (Invoke-VSCode -Command "--list-extensions --show-versions") -Split [Environment]::NewLine
+
+        foreach ($extension in $extensionList)
+        {
+            $info = $extension -Split '@'
+            $extensionName = $info[0]
+            $extensionVersion = $info[1]
+            $currentState.InstalledExtensions[$extensionName] = $extensionVersion
+        }
+
+        foreach ($extension in $currentState.InstalledExtensions.Keys)
+        {
+            if ($extension -eq $this.Name)
+            {
+                if ($null -ne $this.Version)
```

Comment:
`Version` is not a key and therefore does not drive the Present/Absent decision for `Get`.  `Get` uses only keys to find the resource, then returns the other properties set with the values that would best reflect what should be passed to `Test` to return `$true` or `Set` to produce the current state.

Reply:
Updated so that Version does not affect the Present/Absent state. Version match is now checked in the Test() method